### PR TITLE
backend/fix : #354 added unique mobile no records for profile/default…

### PR DESCRIPTION
**Pull request by Dwibedi-10**
…EmergencyNumbers api

## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added a check to insert only unique records for emergencyContacts based on mobile number and also fixed throwing of 500 Internal Server Error when user is trying to enter same mobile number for emergencycontacts.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?
Tested locally.

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by srid** [comment]
@Dwibedi-10 Please merge `main` onto your branch. That should fix the CI failure.

**Comment by Dwibedi-10** [comment]
> @Dwibedi-10 Please merge `main` onto your branch. That should fix the CI failure.

sure

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs [hunk 1]
```diff
@@ -178,3 +180,7 @@ getDefaultEmergencyNumbers personId = do
   personENList <- runInReplica $ QPersonDEN.findAllByPersonId personId
   decPersonENList <- decrypt `mapM` personENList
   return . GetProfileDefaultEmergencyNumbersResp $ DPDEN.makePersonDefaultEmergencyNumberAPIEntity <$> decPersonENList
+
+getUniquePersonByMobileNumber :: UpdateProfileDefaultEmergencyNumbersReq -> [PersonDefaultEmergencyNumber]
+getUniquePersonByMobileNumber req =
+  nubBy (\x y -> mobileNumber x == mobileNumber y) (req.defaultEmergencyNumbers)
```
Comment (srid) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs:186]: Maybe:

```haskell
nubBy ((==) `on` mobileNumber) req.defaultEmergencyNumbers
```

https://hackage.haskell.org/package/base-4.18.0.0/docs/Data-Function.html#v:on

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs [hunk 2]
```diff
@@ -85,7 +86,7 @@ data PersonDefaultEmergencyNumber = PersonDefaultEmergencyNumber
     mobileCountryCode :: Text,
     mobileNumber :: Text
   }
-  deriving (Generic, ToJSON, FromJSON, ToSchema)
+  deriving (Generic, ToJSON, FromJSON, ToSchema, Eq)
```
Comment (roman-bodavskiy) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs:89]: Why do you need `Eq` instance here if you compare by `mobileNumber`?

**Review by roman-bodavskiy — APPROVED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs (hunk 2 above)
Comment (Dwibedi-10) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs]: > Why do you need `Eq` instance here if you compare by `mobileNumber`?

removed it.

**Review by hkmangla — APPROVED** [review]

**Pull request merged**